src/mono/wasi/Wasi.Build.Tests/BuildThenRun.cs

```diff
@@ -13,113 +13,90 @@
-        int expectedExitCode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? 1 : 42;
-
-        string runArgs = $"run --no-build -c {config}";
-        runArgs += " x y z";
-        var res = new RunCommand(s_buildEnv, _testOutput, label: id)
-                            .WithWorkingDirectory(_projectDir!)
-                            .ExecuteWithCapturedOutput(runArgs)
-                            .EnsureExitCode(expectedExitCode);
-
-        Assert.Contains("Hello, Wasi Console!", res.Output);
-        Assert.Contains("args[0] = x", res.Output);
-        Assert.Contains("args[1] = y", res.Output);
-        Assert.Contains("args[2] = z", res.Output);
-
-        if (!_buildContext.TryGetBuildFor(buildArgs, out BuildProduct? product))
-            throw new XunitException($"Test bug: could not get the build product in the cache");
-
-        File.Move(product!.LogFile, Path.ChangeExtension(product.LogFile!, ".first.binlog"));
-
-        _testOutput.WriteLine($"{Environment.NewLine}Publishing with no changes ..{Environment.NewLine}");
-
-        bool expectRelinking = config == "Release";
-        BuildProject(buildArgs,
-                    id: id,
-                    new BuildProjectOptions(
-                        DotnetWasmFromRuntimePack: !expectRelinking,
-                        CreateProject: false,
-                        Publish: true,
                         TargetFramework: BuildTestBase.DefaultTargetFramework,
                         UseCache: false));
-    }
 
-    public static TheoryData<string, bool, bool> TestDataForConsolePublishAndRun()
-    {
-        var data = new TheoryData<string, bool, bool>();
-        data.Add("Debug", false, false);
-        data.Add("Debug", true, true);
-        data.Add("Release", false, false); // Release relinks by default
-        return data;
+        CommandResult res = new RunCommand(s_buildEnv, _testOutput)
+                                    .WithWorkingDirectory(_projectDir!)
+                                    .ExecuteWithCapturedOutput($"run --no-silent --no-build -c {config} x y z")
+                                    .EnsureSuccessful();
+
+        Assert.Contains("Hello, Wasi Console!", res.Output);
     }
 
-    [ConditionalTheory(typeof(BuildTestBase), nameof(IsUsingWorkloads))]
-    [MemberData(nameof(TestDataForConsolePublishAndRun))]
-    public void ConsolePublishAndRunForSingleFileBundle(string config, bool relinking, bool invariantTimezone)
+    [Theory]
+    [InlineData("Debug", /*aot*/ false, /*singleFileBundle*/ false)]
+    [InlineData("Debug", /*aot*/ true, /*singleFileBundle*/ false)]
+    [InlineData("Release", /*aot*/ false, /*singleFileBundle*/ false)]
+    [InlineData("Release", /*aot*/ true, /*singleFileBundle*/ false)]
+    [InlineData("Debug", /*aot*/ false, /*singleFileBundle*/ true)]
+    // [InlineData("Debug", /*aot*/ true, /*singleFileBundle*/ true)] // https://github.com/dotnet/runtime/issues/95273
```

Comment:
I merged a fix for this today.

Reply:
Interesting. I merged with main, including your fixing PR, then I rebuilt and tested: fails with the same error.